packages/sanity/src/hydration-test/event-listener-test.tsx

```diff
@@ -51,23 +48,22 @@ export const eventListenerTest = (): void => {
           ReactDOM.unmountComponentAtNode(root);
         });
 
-        afterEach(() => {
-          eventReset();
-        });
-
         componentMetadata.simulations.forEach((simulation) => {
           it('component should unmount without leaving event listeners on the window, document, and body', () => {
-            const {windowEe, documentEe, bodyEe, reset} = overrideEventListeners();
-            eventReset = reset;
+            const {windowLogger, documentLogger, bodyLogger} = overrideEventListeners();
```

Comment:
I would unwrap the function and create the loggers here,
```
const windowLogger = attachEventListenerLogger(window);
const documentLogger = attachEventListenerLogger(document);
const bodyLogger = attachEventListenerLogger(document.body);
```
... two extra lines, but it would nicely mirror those 3 detach and assert calls.